src/main/kotlin/no/nav/k9/los/nyoppgavestyring/query/db/OppgaveQueryRepository.kt

```diff
@@ -0,0 +1,105 @@
+package no.nav.k9.los.nyoppgavestyring.query.db
+
+import kotliquery.TransactionalSession
+import kotliquery.queryOf
+import kotliquery.sessionOf
+import kotliquery.using
+import no.nav.k9.los.nyoppgavestyring.query.*
+import javax.sql.DataSource
+
+class OppgaveQueryRepository(val datasource: DataSource) {
+
+    fun hentAlleFelter(): Oppgavefelter {
+        return using(sessionOf(datasource)) { it ->
+            it.transaction { tx -> Oppgavefelter(hentAlleFelter(tx)) }
+        }
+    }
+
+    private fun midlertidigFiksVisningsnavn(kode: String): String {
+        val s = kode.replace("([A-ZÆØÅ])".toRegex(), " $1").lowercase();
+        return s.substring(0, 1).uppercase() + s.substring(1)
+    }
+
+    private fun hentAlleFelter(tx: TransactionalSession): List<Oppgavefelt> {
+        val felterFraDatabase = tx.run(
+            queryOf(
+                """
+                    SELECT DISTINCT fo.ekstern_id as omrade,
+                      fd.ekstern_id as kode,
+                      fd.ekstern_id as visningsnavn,
+                      fd.tolkes_som as tolkes_som
+                    FROM Feltdefinisjon fd INNER JOIN Omrade fo ON (
+                      fo.id = fd.omrade_id
+                    )
+                """.trimIndent()
+            ).map{row -> Oppgavefelt(
+                row.string("omrade"),
+                row.string("kode"),
+                midlertidigFiksVisningsnavn(row.string("visningsnavn")),
+                row.string("tolkes_som")
+            ) }.asList
+        ) ?: throw IllegalStateException("Feil ved kjøring av hentAlleFelter")
+
+        val standardfelter = listOf(
+            Oppgavefelt(null, "oppgavestatus", "Oppgavestatus", "String"),
+            Oppgavefelt(null, "kildeområde", "Kildeområde", "String"),
+            Oppgavefelt(null, "oppgavetype", "Oppgavetype", "String"),
+            Oppgavefelt(null, "oppgaveområde", "Oppgaveområde", "String")
+        )
+
+        return (felterFraDatabase + standardfelter).sortedBy { it.visningsnavn };
+    }
+
+    fun query(oppgaveQuery: OppgaveQuery): List<Long> {
+        return using(sessionOf(datasource)) { it ->
+            it.transaction { tx -> query(tx, oppgaveQuery) }
```

Comment:
Genererer opp transaksjoner både her og i OppgaveQueryService. Konsolider?

Forøvrig, er det egentlig noe poeng med transaksjoner på disse spørringene i det hele tatt? Like bra med bare Session.run() ? Se OppgaveV3Repository.tellAntall() for eksempel